recipes/implot/all/test_package/conanfile.py

```diff
@@ -1,15 +1,31 @@
-from conans import ConanFile, CMake, tools
 import os
 
-class TestPackageConan(ConanFile):
+from conan import ConanFile
+from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout, CMakeDeps
+from conan.tools.build import can_run
+
+
+class ImplotTestConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake", "cmake_find_package_multi"
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+
+    def generate(self):
+        deps = CMakeDeps(self)
+        deps.generate()
+        tc = CMakeToolchain(self)
+        tc.generate()
 
     def build(self):
         cmake = CMake(self)
         cmake.configure()
         cmake.build()
 
+    def layout(self):
+        cmake_layout(self)
+
     def test(self):
-        if not tools.cross_building(self):
-            self.run(os.path.join("bin", "test_package"), run_environment=True)
+        if can_run(self):
+            cmd = os.path.join(self.cpp.build.bindir, "test_package")
```

Comment:
```suggestion
            cmd = os.path.join(self.cpp.build.bindirs[0], "test_package")
```

Reply:
I'm not sure if this is needed. At least conan 2.0.2 just has `bindir` and not `bindirs[0]` in its template for `conan new cmake_lib`.

Reply:
Yes you're right. Seems they're equivalent:

https://docs.conan.io/en/1.59/migrating_to_2.0/recipes.html#cpp-info-libdir-bindir-includedir-accessors-when-using-layout-in-conan-1-x

> Since [Conan 1.53.0](https://github.com/conan-io/conan/releases/tag/1.53.0) you can access cpp_info.libdirs[0], cpp_info.bindirs[0] and cpp_info.includedirs[0] using cpp_info.libdir, cpp_info.bindir and cpp_info.includedir

But I would still proceed with caution because this is not documented in v2 cpp_info section https://docs.conan.io/2/reference/conanfile/methods/package_info.html#cpp-info-library-and-build-information

Reply:
our template need to be updated 😱